list.js

```diff
@@ -1,18 +1,27 @@
 (function() {
-    var events = [];
+    var eventsMap = new Map();
 
-    var renderEvents = function(events) {
+    var renderAllEvents = function(eventList) {
         var markup = '';
+        eventList.forEach(function(value, key, map) {
+            markup += value.render();
```

Comment:
So, I believe that this is actually OK in modern browsers as they basically optimize for string concatenation, but it most other languages/environments, you would want to avoid concatenating strings in a loop. Eg, in Python strings are immutable, so it turns it into an n^2 operation and you would be much better off building an array of strings and then doing a join() on it.

Anyway, it still kind of weirds me out whenever I see string concatenation in a loop in JS...